### PR TITLE
feat: support Vanilla Linux

### PR DESCRIPTION
**Pull request by SteveLauC**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] The code compiles (`cargo build`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [ ] *Optional:* I have tested the code myself
    - [ ] I also tested that Topgrade skips the step where needed

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

----
Ref:

* [manual](https://github.com/Vanilla-OS/apx/blob/fd7a42edbe5d3f9d251267f51211648394eddf45/man/man1/apx.1#L697-L713) for `apx update`
* [manual](https://github.com/Vanilla-OS/apx/blob/fd7a42edbe5d3f9d251267f51211648394eddf45/man/man1/apx.1#L660-L670) for `apx upgrade`

They do have a [rendered online manual page](https://documentation.vanillaos.org/docs/apx/manpage), but it is kinda outdated.

### CODE REVIEW

File: src/steps/os/linux.rs [hunk 1]
```diff
@@ -74,6 +76,11 @@ impl Distribution {
             Some("openmandriva") => Distribution::OpenMandriva,
             Some("pclinuxos") => Distribution::PCLinuxOS,
             _ => {
+                if let Some(name) = name {
```
Comment (SteveLauC) [inline src/steps/os/linux.rs:79]: They don't have a unique `ID` for now, so we have to check the `NAME` field, but they will have one when they reach version 2.0, and we can preview it [here](https://github.com/Vanilla-OS/base-files/blob/2414c26f325e78295ee6e59a89b199fdd7d8f850/etc/os-release#L6).

**Comment by SteveLauC** [comment]
Don’t merge this for now, let me find some folk from the Vanilla Community to get it reviewed first

**Review by DottoDev — APPROVED** [review]

**Comment by SteveLauC** [comment]
Emmm, alright then, I honestly think this patch should work as I have tested the following commands:

```shell
$ apx update --all -y
$ apx upgrade --all -y
```
in a live boot.

**Comment by DottoDev** [comment]
> Don’t merge this for now, let me find some folk from the Vanilla Community to get it reviewed first

Was on my phone and your comment didn't show up xD

Should I revert the merge?

**Comment by SteveLauC** [comment]
> Should I revert the merge?

Let's leave it merged, I think this is generally fine. 

> Don’t merge this for now, let me find some folk from the Vanilla Community to get it reviewed first

And probably there won't be any folk from the Vanilla community that will give it a review, sigh.  I will try to set up an installed Vanilla OS in a VM tomorrow, and give this patch a full test. 
Vanilla OS will roughly take an hour to install, and I have no idea if it is stuck or still installing during the installation, I hope I can get it successfully installed tomorrow...

**Pull request merged**